web/satellite/src/components/dashboard/AddUserPopup.vue

```diff
@@ -0,0 +1,423 @@
+
+@Component(
+    {
+        props: {
+            onClose: Function
+        },
+        data: function() {
+            return {
+                inputs: [new EmailInput(), new EmailInput(), new EmailInput()],
+                formError: '',
+                imageSource: EMPTY_STATE_IMAGES.ADD_USER,
+                imageDeleteUser: EMPTY_STATE_IMAGES.DELETE_USER,
+            };
+        },
+        methods: {
+            onAddUsersClick: async function() {
+                let length = this.$data.inputs.length;
+                let newInputsArray: any[] = [];
+                let areAllEmailsValid = true;
+                let emailArray: string[] = [];
+
+                for (let i = 0; i < length; i++) {
+                    let element = this.$data.inputs[i];
+                    let isEmail = validateEmail(element.value);
+
+                    if (isEmail) {
+                        emailArray.push(element.value);
+                    }
+
+                    if (isEmail || element.value === '') {
+                        element.setError(false);
+                        newInputsArray.push(element);
+
+                        continue;
+                    }
+
+                    element.setError(true);
+                    newInputsArray.unshift(element);
+                    areAllEmailsValid = false;
+
+                    this.$data.formError = 'Field is required. Please enter a valid email address';
+                }
+
+                this.$data.inputs = newInputsArray;
+
+                if (length > 3) {
+                    let scrollableDiv: any = document.querySelector('.add-user__form-container__inputs-group');
+
+                    if (scrollableDiv) {
+                        let scrollableDivHeight = scrollableDiv.offsetHeight;
+                        scrollableDiv.scroll(0, -scrollableDivHeight);
+                    }
+                }
+
+                if (!areAllEmailsValid) return;
+                
+                let result = await this.$store.dispatch('addProjectMembers', emailArray);
+                
+                if (!result.isSuccess) {
+                    this.$store.dispatch('error', 'Error during adding team members!');
+
+                    return;
+                }
+
+                const response = await this.$store.dispatch('fetchProjectMembers', { limit: 20, offset: 0 });
+
+                if (!response.isSuccess) {
+                    this.$store.dispatch('error', 'Unable to fetch project members');
+            
+                    return;
+                }
+
+                this.$store.dispatch('success', 'Members successfully added to project!');
+                this.$store.dispatch('setAddTeamMembersPopup', false);
+            },
+            addInput: function(): void {
+                let inputsLength = this.$data.inputs.length;
+
+                if (inputsLength < 10) {
+                    this.$data.inputs.push(new EmailInput());
+                }
```

Comment:
`if (this.$data.inputs.length > 9) {
                    return;
                }

                this.$data.inputs.push(new EmailInput());`